src/provisioner/validators.py

```diff
@@ -0,0 +1,199 @@
+        for id_key in nodes_in_conf_store.keys():
+            node_info = nodes_in_conf_store[id_key]
+            storage_set_id = node_info['storage_set']
+            if storage_set_in_conf_store.get(storage_set_id) is None:
+                storage_set_in_conf_store[storage_set_id] = []
+            storage_set_in_conf_store[storage_set_id].append(node_info)
+
+        # validate nodes define for each storage_set in conf_store,
+        # is same as nodes define in solution_config
+        for key in storage_set_in_conf_store.keys():
+            CS_nodes = storage_set_in_conf_store[key]
+            SC_nodes = storage_set_in_sol_conf[key]
+            if len(CS_nodes) != len(SC_nodes):
+                raise CortxProvisionerError(errno.EINVAL,
+                    'No of nodes define in conf_store is not same as solution_config.'
+                    f' Nodes in {self.solution_conf_url} is {len(SC_nodes)} and '
+                    f'in {self.cortx_conf_url} is {len(CS_nodes)}')
+
+        return 0
+
+    def check_external_services(self):
+        """" Validate no of external services in cortx config """
+
+        sol_conf_ext_services = self.get_value_from_solution_config(
+            'cortx>external')
+        conf_stor_ext_services = self.get_value_from_conf_store(
+            'cortx>external')
+
+        if len(conf_stor_ext_services) != len(sol_conf_ext_services):
+            raise CortxProvisionerError(errno.EINVAL,
+                'External services define in conf_store is not same as '
+                f'solution_config. External services in {self.solution_conf_url}'
+                f' is {len(sol_conf_ext_services)} and in {self.cortx_conf_url}'
+                f' is {len(conf_stor_ext_services)}')
+
+        # Check endpoints
+        for service in conf_stor_ext_services.keys():
+            if (conf_stor_ext_services[service]['endpoints'] !=
+                    sol_conf_ext_services[service]['endpoints']):
+                raise CortxProvisionerError(errno.EINVAL,
+                    f'{service} endpoints define in conf_store is not same as '
+                    'solution_config.')
+        return 0
+
+    def get_value_from_solution_config(self, key):
+        """ Read config value for key from solution_config """
+
+        config_value = Conf.get(
+            ConfigValidator._solution_index, key)
+        if config_value is None:
+            raise CortxProvisionerError(
+                errno.EINVAL,
+                f'{key} key is unspecified for {self.solution_conf_url}')
+
+        return config_value
+
+    def get_value_from_conf_store(self, key):
+        """ Read config value from conf_store """
+
+        config_value = self.cortx_config_store.get(key)
+        if config_value is None:
+            raise CortxProvisionerError(
+                errno.EINVAL,
+                f'{key} key is unspecified for {self.cortx_conf_url}')
+
+        return config_value
+
+
+if __name__ == "__main__":
```

Comment:
Remove this main. You can call this as just like any other module
import validators.validator
validator.validate(['what'], args...)